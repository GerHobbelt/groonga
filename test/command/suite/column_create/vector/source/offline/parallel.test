#@require-apache-arrow
#$GRN_TOKEN_COLUMN_PARALLEL_TABLE_SIZE_THRESHOLD=1
#$GRN_TOKEN_COLUMN_PARALLEL_CHUNK_SIZE=2

table_create Terms TABLE_PAT_KEY ShortText \
  --normalizer NormalizerNFKC121 \
  --default_tokenizer TokenNgram

table_create Notes TABLE_NO_KEY
column_create Notes title COLUMN_SCALAR Text

load --table Notes
[
{"title": "Hello Groonga"},
{"title": "Hello Mroonga"},
{"title": "Hello Rroonga"}
]

column_create Notes title_terms COLUMN_VECTOR Terms title

column_create Terms notes_title COLUMN_INDEX|WITH_POSITION Notes title

dump

index_column_diff Terms notes_title
