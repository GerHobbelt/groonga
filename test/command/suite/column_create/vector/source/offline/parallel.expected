table_create Terms TABLE_PAT_KEY ShortText   --normalizer NormalizerNFKC121   --default_tokenizer TokenNgram
[[0,0.0,0.0],true]
table_create Notes TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Notes title COLUMN_SCALAR Text
[[0,0.0,0.0],true]
load --table Notes
[
{"title": "Hello Groonga"},
{"title": " "},
{"title": "Hello Mroonga"},
{"title": "Hello Rroonga"}
]
[[0,0.0,0.0],4]
column_create Notes title_terms COLUMN_VECTOR Terms title
[[0,0.0,0.0],true]
select Notes   --query _id:2   --output_columns title_terms
[[0,0.0,0.0],[[[1],[["title_terms","Terms"]],[[]]]]]
dump
table_create Notes TABLE_NO_KEY
column_create Notes title COLUMN_SCALAR Text

table_create Terms TABLE_PAT_KEY ShortText --default_tokenizer TokenNgram --normalizer NormalizerNFKC121

column_create Notes title_terms COLUMN_VECTOR Terms title

load --table Terms
[
["_key"],
["groonga"],
["hello"],
["mroonga"],
["rroonga"]
]

load --table Notes
[
["_id","title"],
[1,"Hello Groonga"],
[2," "],
[3,"Hello Mroonga"],
[4,"Hello Rroonga"]
]
column_create Terms notes_title COLUMN_INDEX|WITH_POSITION Notes title
[[0,0.0,0.0],true]
index_column_diff Terms notes_title
[[0,0.0,0.0],[]]
